cmd/kubeadm/app/cmd/join.go

```diff
@@ -342,6 +342,9 @@ func newJoinData(cmd *cobra.Command, args []string, opt *joinOptions, out io.Wri
 
 	// if not joining a control plane, unset the ControlPlane object
 	if !opt.controlPlane {
+		if len(opt.externalcfg.ControlPlane.CertificateKey) > 0 {
+			klog.Warningf("[preflight] WARNING: %s will be ignored when %s is not set.", options.CertificateKey, options.ControlPlane)
+		}
```

Comment:
long term, we should not bind more logic related to flags and flags mixtures.
some day, ideally, we should move all flags to configuration-only and configuration patching.